### PR TITLE
Disallow unarchiving catalogue items if dependencies are archived

### PR DESCRIPTION
**Pull request by okahilak**

### CODE REVIEW

File: test/clj/rems/api/services/test_catalogue.clj [hunk 1]
```diff
@@ -66,7 +92,40 @@
...
+      (is (:success (archive-catalogue-item! true))))
+
+    (testing "cannot unarchive if workflow is archived"
+      (archive-catalogue-item! true)
+      (archive-workflow! true)
+      (is (not (:success (archive-catalogue-item! false))))
+      (archive-workflow! false)
+      (is (:success (archive-catalogue-item! true))))
+
+    (testing "cannot unarchive if license is archived"
```
Comment (opqdonut) [inline test/clj/rems/api/services/test_catalogue.clj]: resource and license?

File: test/clj/rems/api/services/test_catalogue.clj [hunk 2]
```diff
@@ -66,7 +92,40 @@
...
+      (archive-catalogue-item! true)
+      (archive-resource! true)
+      (archive-license! true)
+      (let [errors (:errors (archive-catalogue-item! false))]
+        (is (= (set (mapv :type errors))
+               #{:t.administration.errors/resource-archived
+                 :t.administration.errors/license-archived})))
```
Comment (opqdonut) [inline test/clj/rems/api/services/test_catalogue.clj]: nit: expected value comes first

File: src/clj/rems/api/services/catalogue.clj [hunk 3]
```diff
@@ -16,9 +28,25 @@
     return))
 
 (defn update-catalogue-item! [command]
-  ;; TODO disallow unarchiving catalogue item if its resource, form or licenses are archived
-  (db/set-catalogue-item-state! (select-keys command [:id :enabled :archived]))
-  {:success true})
+  (let [{:keys [form resource workflow licenses]} (dependencies-for-catalogue-item (:id command))
+        archived-licenses (filter #(:archived %) licenses)
```
Comment (opqdonut) [inline src/clj/rems/api/services/catalogue.clj]: AKA `(filter :archived licenses)`

File: src/clj/rems/api/services/catalogue.clj [hunk 4]
```diff
@@ -16,9 +28,25 @@
...
+         [(when (:archived form)
+            {:type :t.administration.errors/form-archived :forms [form]})
+          (when (:archived resource)
+            {:type :t.administration.errors/resource-archived :resources [resource]})
+          (when (:archived workflow)
+            {:type :t.administration.errors/workflow-archived :workflows [workflow]})
+          (when (not (empty? archived-licenses))
+            {:type :t.administration.errors/license-archived :licenses archived-licenses})])]
+    (if (and (not (:archived command))
+             (pos? (count errors)))
```
Comment (opqdonut) [inline src/clj/rems/api/services/catalogue.clj]: AKA `(not (empty? errors))`

File: test/clj/rems/api/services/test_catalogue.clj (hunk 1 above)
Comment (okahilak) [reply test/clj/rems/api/services/test_catalogue.clj]: Done.

File: test/clj/rems/api/services/test_catalogue.clj (hunk 2 above)
Comment (okahilak) [reply test/clj/rems/api/services/test_catalogue.clj]: Done.

File: src/clj/rems/api/services/catalogue.clj (hunk 3 above)
Comment (okahilak) [reply src/clj/rems/api/services/catalogue.clj]: Done.

File: src/clj/rems/api/services/catalogue.clj (hunk 4 above)
Comment (okahilak) [reply src/clj/rems/api/services/catalogue.clj]: Done.

**Pull request merged**